src/ip_notifier.cpp

```diff
@@ -54,11 +54,29 @@ namespace libtorrent { namespace aux
 {
 namespace
 {
+template <typename...>
+struct ip_change_notifier_none final : ip_change_notifier
+{
+	explicit ip_change_notifier_none(io_service& ios) : m_ios(ios) {}
+
+	void async_wait(std::function<void(error_code const&)> cb) override
+	{
+		m_ios.post([cb]()
+		{ cb(make_error_code(boost::system::errc::not_supported)); });
+	}
+
+	void cancel() override {}
+
+private:
+	io_service& m_ios;
+};
+
 #if defined TORRENT_BUILD_SIMULATOR
-	// TODO: ip_change_notifier_sim
+using ip_change_notifier_sim = ip_change_notifier_none<>;
```

Comment:
it seems it would be simpler if all of these platform proprocessor blocks would produce the same name of the ip_change_notifier type, or at least provide an alias with the same name. That way you wouldn't need to repeat the if-chain below, to pick the right one to construct.